lib/cc/analyzer/include_paths_builder.rb

```diff
@@ -17,152 +14,76 @@ def initialize(cc_exclude_paths, cc_include_paths = [])
       end
 
       def build
-        if cc_include_paths.any?
-          paths = filter_by_cc_includes
-        else
-          root = Directory.new(".", ignored_files)
-          paths = root.included_paths
-        end
+        @_build ||= PathMinimizer.new(paths).minimize
+      end
 
-        paths.each do |path|
-          raise_on_unreadable_files(path)
-        end
+      def build_with_excludes(excludes)
+        new_paths = paths.reject { |path| matches_globs?(path, excludes) }
+        PathMinimizer.new(new_paths).minimize
       end
```

Comment:
I think having both a `build` and `build_with_excludes` sort of muddies the intention of this class. And, in fact, you could argue that `build` here is just `build_with_excludes` with an empty excludes array. What do you think about merging the two?